File: clients/js/src/haveapi/validators/format.js
Validator.validators.format = function (opts, value) {
	if (typeof value != 'string')
		return false;

	var rx = new RegExp(opts.rx);

	if (opts.match)
		return value.match(rx) ? true : false;

	return value.match(rx) ? false : true;
};

File: clients/ruby/lib/haveapi/client/validators/format.rb
module HaveAPI::Client
  class Validators::Format < Validator
    name :format

    def valid?
      rx = Regexp.new(opts[:rx])

      if opts[:match]
        rx.match(value) ? true : false

      else
        rx.match(value) ? false : true
      end
    end
  end
end
